tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -80,6 +80,17 @@ public static void Initialize()
                 Log.Error(ex, "ByRef instrumentation cannot be enabled: ");
             }
 
+            try
+            {
+                Log.Debug("Enabling calltarget state by ref.");
+                NativeMethods.EnableCallTargetStateByRef();
```

Comment:
Am I right in thinking this completely avoids the situation where `enable_calltarget_state_by_ref` is `true`, but the managed integrations don't expect a by ref argument?

For example, if there's an exception here, that seems like a fatal problem, as we would have a mismatch for jit rewriting? We should disable instrumentation entirely in that case shouldn't we? i.e. if this (or the existing `EnableByRefInstrumentation`) calls fail, should we bail out of this method entirely as we can't safely instrument?

Reply:
If this call fails, the rewriting will not emit the calltargetstate by ref, so it will be like the previous approach.

If that is the case then these`EndMethods` will be called instead the newer ones: https://github.com/DataDog/dd-trace-dotnet/blob/tony/calltargetstate-by-ref/tracer/src/Datadog.Trace/ClrProfiler/CallTarget/CallTargetInvoker.cs#L509-L549 converting the previous method to a `byref` call. (line 525 and 545)

Reply:
The thing is we are still handling both cases, so we can still use this same managed integrations in an older profiler without the byref rewriting.

Reply:
>The thing is we are still handling both cases

Ah cool, I missed that point, I thought we were changing to always use `byref`. That's all good then 🙂